test/controllers/posts_controller_test.rb

```diff
@@ -0,0 +1,13 @@
+require 'test_helper'
+
+class PostsControllerTest < ActionDispatch::IntegrationTest
+  test 'should get index' do
+    get posts_index_url
+    assert_response :success
+  end
+
+  test 'should get show' do
+    get posts_show_url
+    assert_response :success
+  end
+end
```

Comment:
- [ ] What you have written is the old method and it is not recommended for new application. `Using the older approach of mocking out the whole framework with ActionDispatch::ControllerTest is no longer recommended for new applications.` and hence in the new approach which is also recommended by Microverse is to  `Create a Request spec file for all your controllers` which I have show below how you can do or you can take reference from [here as well](https://relishapp.com/rspec/rspec-rails/docs/request-specs/request-spec).

- [ ] when you have created the controller, it should have created `spec/requests/` directory with `type: request `  as shown in the example code. If not, you can create these folder manually and inside that you can create two file naming `users_spec.rb` and `posts_spec.rb`.

```ruby

# spec/requests/posts_spec.rb
require 'rails_helper'

 RSpec.describe "Posts", type: :request do
   ...
 end
```

and inside `Rspec.describe` you can write `describe` as shown below and inside `describe` you can write `it` block and test the controller.

```ruby
describe 'GET /index' do
  ...
end

describe 'GET /show' do
  ...
end
```

- [ ] Spec file include tests that ensures correct view template is rendered. Example : `expect(response).to render_template(:show)`

- [ ] Spec file include tests that ensures the response body includes the correct placeholder text. Example : `expect(response.body).to include("Posts#index")`